### PR TITLE
Add the exponential backoff wait if fail attemps on the arduino end

### PR DESCRIPTION
**Pull request by hbgxsm95**
Add the exponential backoff wait feature on the arduino end. If too many fail attempts,  the boundary of wait time would not change and turn on the LED for notification. When receiving the positive response, refresh the counter.

### CODE REVIEW

File: go/iot/protocol/examples/arduino/protocol.ino [hunk 1]
```diff
@@ -75,11 +81,27 @@ void request(void (*func)()) {
...
-        delay(1000);
+        if (failtime == FAILTHRESHOLD){
+          // When there are too many fail attempts, turn on the light to notify
+          // Meanwhile stop increasing the fail attempts counter
+          digitalWrite(LED_BUILTIN, HIGH);
+        }
+        else{
+          failtime = failtime + 1;
+        }
+        delay(random(1 << failtime));
```
Comment (songy23) [inline go/iot/protocol/examples/arduino/protocol.ino]: I don't think you need `random` here. IIUC exponential backoff should be deterministic.

Comment (hbgxsm95) [reply go/iot/protocol/examples/arduino/protocol.ino]: But according to the https://en.wikipedia.org/wiki/Exponential_backoff, in the protocol of Ethernet, the exponential time should be random from a range of time slots.

Comment (songy23) [reply go/iot/protocol/examples/arduino/protocol.ino]: I see, then consider adding the random number as a decimal. For example https://cloud.google.com/storage/docs/exponential-backoff#example_algorithm.

Comment (hbgxsm95) [reply go/iot/protocol/examples/arduino/protocol.ino]: Okay.

**Pull request merged**